### ISSUE TITLE
boolean request content does not compile

### ISSUE THREAD

**Issue by cataggar** [Azure/autorest.csharp#3639]
I'm switching over an internal API from Swagger to TypeSpec in this [internal PR](https://dev.azure.com/msazure/One/_git/Azure-Dedicated-AVS/pullrequest/8554320). The DPG code generated for operation `suppressSubscriptionMetering` does not compile. 

``` tsp
@route("/v1/subscription/{subscriptionId}/metering/suppress")
@put
op suppressSubscriptionMetering(
  @doc("Azure Subscription ID")
  @path
  subscriptionId: string,

  @body body: boolean,
): void;
```

Generates:
``` cs
        public virtual async Task<Response> SuppressSubscriptionMeteringAsync(string subscriptionId, bool body, CancellationToken cancellationToken = default)
        {
            Argument.AssertNotNullOrEmpty(subscriptionId, nameof(subscriptionId));

            RequestContext context = FromCancellationToken(cancellationToken);
            Response response = await SuppressSubscriptionMeteringAsync(subscriptionId, body, context).ConfigureAwait(false);
            return response;
        }
```

Which fails with:
C:\Users\cataggar\ms\avs\src\conveyor\clients\dotnet\Generated\ConveyorApiClient.cs(10921,95): error CS1503: Argument 3: cann
ot convert from 'Azure.RequestContext' to 'System.Threading.CancellationToken' [C:\Users\cataggar\ms\avs\src\conveyor\clients 
\dotnet\ConveyorApi.csproj::TargetFramework=net6.0]

One manual workaround is to use `cantext.CancellationToken`.

"@typespec/compiler": "^0.46.0",
Microsoft.Azure.AutoRest.CSharp 3.0.0-beta.20230804.1

**Comment by pshao25** [comment]
We don't support boolean as a body. Do you have any reason the body must be a boolean?

To unblock you could try to make body as a model with a boolean property?

**Comment by cataggar** [comment]
Thanks @pshao25. I'll see if we can get the API changed.

**Comment by cataggar** [comment]
Similar to https://github.com/Azure/autorest.csharp/issues/3667, this would work if the generated code was changed to be like:
``` cs
        public virtual async Task<Response> SuppressSubscriptionMeteringAsync(string subscriptionId, bool body, CancellationToken cancellationToken = default)
        {
            Argument.AssertNotNullOrEmpty(subscriptionId, nameof(subscriptionId));

            RequestContext context = FromCancellationToken(cancellationToken);
            // Response response = await SuppressSubscriptionMeteringAsync(subscriptionId, body, context).ConfigureAwait(false);
           Response response = await SuppressSubscriptionMeteringAsync(subscriptionId, JsonRequestContent.FromBool(body), context).ConfigureAwait(false);
            return response;
        }
```

**Comment by pshao25** [comment]
@cataggar please provide the justification why the body should be a `boolean` instead of a model with `boolean` property as suggested above. 

And have your justification been reviewed by API review team?

**Comment by cataggar** [comment]
This is not an ARM API.

**Comment by mikekistler** [comment]
It is not a goal to support every API that can be defined in TypeSpec with DPG -- only APIs that follow the Azure REST API Guidelines or are already public GA Azure services.
